java/lib.jshell.agent/agentsrc/org/netbeans/lib/jshell/agent/AgentWorker.java

```diff
@@ -462,11 +462,12 @@ private String performExecute(String className, String methodName, CountDownLatc
                 userCodeExecutingThreads.put(socketPort, Thread.currentThread());
             }
             return super.invoke(className, methodName);
-        } catch (ThreadDeath td) {
-            LOG.log(Level.FINE, "Received ThreadDeath, killed: {0}", killed);
-            if (!killed.get()) {
-                throw td;
-            }
+        // TODO this code relied on performStop() calling Thread.stop()
+//        } catch (ThreadDeath td) {
+//            LOG.log(Level.FINE, "Received ThreadDeath, killed: {0}", killed);
+//            if (!killed.get()) {
+//                throw td;
+//            }
```

Comment:
I wasn't able to reach this code path. But I also don't really know all usecases of the jshell integration, outside of opening it via the tools menu.

But since nobody complained about `UnsupportedOperationException`s so far, it might be very niche or not reachable anymore at this point.